nsxt/resource_nsxt_upgrade_precheck_acknowledge.go

```diff
@@ -0,0 +1,96 @@
+	nsxModel "github.com/vmware/vsphere-automation-sdk-go/services/nsxt-mp/nsx/model"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt-mp/nsx/upgrade"
+)
+
+func resourceNsxtUpgradePrecheckAcknowledge() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNsxtUpgradePrecheckAcknowledgeCreate,
+		Read:   resourceNsxtUpgradePrecheckAcknowledgeRead,
+		Update: resourceNsxtUpgradePrecheckAcknowledgeUpdate,
+		Delete: resourceNsxtUpgradePrecheckAcknowledgeDelete,
+
+		Schema: map[string]*schema.Schema{
+			"precheck_ids": {
+				Type:        schema.TypeList,
+				Description: "IDs of precheck warnings that need to be acknowledged",
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Required: true,
+			},
+		},
+	}
+}
+
+func resourceNsxtUpgradePrecheckAcknowledgeCreate(d *schema.ResourceData, m interface{}) error {
+	id := d.Id()
+	if id == "" {
+		id = newUUID()
+	}
+	d.SetId(id)
+	precheckIDs := interface2StringList(d.Get("precheck_ids").([]interface{}))
+	err := acknowledgePrecheckWarnings(m, precheckIDs)
+	if err != nil {
+		return handleCreateError("NsxtPrecheckAcknowledge", id, err)
+	}
+	return resourceNsxtUpgradePrecheckAcknowledgeRead(d, m)
+}
+
+func acknowledgePrecheckWarnings(m interface{}, precheckIDs []string) error {
+	connector := getPolicyConnector(m)
+	client := upgrade.NewPreUpgradeChecksClient(connector)
+	for _, precheckID := range precheckIDs {
+		err := client.Acknowledge(precheckID)
+		if err != nil {
+			return fmt.Errorf("Failed to acknowledge precheck warning with ID %s: %s", precheckID, err)
+		}
+	}
+	return nil
+}
+
+func resourceNsxtUpgradePrecheckAcknowledgeRead(d *schema.ResourceData, m interface{}) error {
+	id := d.Id()
+	precheckWarnings, err := getPrecheckErrors(m, nsxModel.UpgradeCheckFailure_TYPE_WARNING)
+	if err != nil {
+		return handleReadError(d, "NsxtUpgradePrecheckAcknowledge", id, err)
+	}
+	err = setAcknowledgedPrecheckIDsInSchema(d, precheckWarnings)
+	if err != nil {
+		return handleReadError(d, "NsxtUpgradePrecheckAcknowledge", id, err)
+	}
+	return nil
+}
+
+func setAcknowledgedPrecheckIDsInSchema(d *schema.ResourceData, precheckWarnings []nsxModel.UpgradeCheckFailure) error {
+	var precheckWarningIDs []string
+	for _, precheckWarning := range precheckWarnings {
+		if !(*precheckWarning.NeedsAck) {
+			id := *precheckWarning.Id
+			precheckWarningIDs = append(precheckWarningIDs, id)
+		}
+	}
+	return d.Set("precheck_ids", precheckWarningIDs)
+}
+
+func resourceNsxtUpgradePrecheckAcknowledgeUpdate(d *schema.ResourceData, m interface{}) error {
+	id := d.Id()
+	precheckIDs := interface2StringList(d.Get("precheck_ids").([]interface{}))
+	err := acknowledgePrecheckWarnings(m, precheckIDs)
+	if err != nil {
+		return handleUpdateError("NsxtPrecheckAcknowledge", id, err)
+	}
+	return resourceNsxtUpgradePrecheckAcknowledgeRead(d, m)
+}
+
+func resourceNsxtUpgradePrecheckAcknowledgeDelete(d *schema.ResourceData, m interface{}) error {
+	return nil
```

Comment:
there is no mechanism to un-acknowledge a precheck? Not sure if it makes sense :)

Reply:
No there is no un-acknowledge operation for precheck warnings(from either API or UI)